contracts/src/l1/TokenBridge.sol

```diff
@@ -170,21 +240,89 @@ contract TokenBridge is
         messagingContract.consumeMessageFromL2(l2TokenBridge, payload);
     }
 
-    function deposit(address l1Token_, uint256 l2Recipient, uint256 amount) onlyApprovedToken(l1Token_) external {
-        IStaticATokenLM l1Token = IStaticATokenLM(l1Token_);
-        l1Token.safeTransferFrom(msg.sender, address(this), amount);
-        sendMessage(l1Token_, msg.sender, l2Recipient, amount);
+    function _dynamicToStaticAmount(
+        uint256 amount,
+        address asset,
+        ILendingPool lendingPool
+    ) internal view returns (uint256) {
+        return amount.rayDiv(lendingPool.getReserveNormalizedIncome(asset));
+    }
+
+    function _staticToDynamicAmount(
+        uint256 amount,
+        address asset,
+        ILendingPool lendingPool
+    ) internal view returns (uint256) {
+        return amount.rayMul(lendingPool.getReserveNormalizedIncome(asset));
+    }
+
+    function deposit(
+        address l1AToken,
+        uint256 l2Recipient,
+        uint256 amount,
+        uint16 referralCode,
+        bool fromAsset
+    ) external onlyApprovedToken(l1AToken) onlyValidL2Address(l2Recipient) {
```

Comment:
check whether underlyingAsset or lendingPool attributes exist instead of onlyApprovedToken(l1AToken)